src/mlpack/core/data/load_image.hpp

```diff
@@ -0,0 +1,144 @@
+/**
+ * @file load_image.hpp
+ * @author Mehul Kumar Nirala
+ *
+ * An image loading utillity
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_CORE_DATA_LOAD_IMAGE_HPP
+#define MLPACK_CORE_DATA_LOAD_IMAGE_HPP
+
+#include <string>
+#include <iostream>
+#include <mlpack/core.hpp>
+#include <mlpack/core/util/log.hpp>
+#include <mlpack/prereqs.hpp>
+#include <boost/filesystem.hpp>
+
+#include "extension.hpp"
+
+#define STB_IMAGE_IMPLEMENTATION
+#include <stb_image.h>
+
+namespace mlpack {
+namespace data {
+
+/**
+ * Loads a matrix with image. It also supports loading image from 
+ * an entire directory.
```

Comment:
Can we provide a simple example here; something like: https://github.com/mlpack/mlpack/blob/584f8533f20a4b94831d6b68ebb100bcbee573b7/src/mlpack/core/data/binarize.hpp#L26-L29

Reply:
Okay sure